pandas/_libs/index.pyx

```diff
@@ -270,7 +270,14 @@ cdef class IndexEngine:
             Py_ssize_t i, j, n, n_t, n_alloc
 
         self._ensure_mapping_populated()
-        values = np.array(self._get_index_values(), copy=False)
+        if targets[isnaobj(targets)].size > 0:
```

Comment:
```suggestion
        if isnaobj(targets)].any():
```

I think the extra indexing step here is unnecessary, no?

Reply:
Agreed, I will update.